### PR TITLE
Website: demo fixes (Safari)

### PR DESCRIPTION
**Pull request by 1chandu**
For #976
- Fix Safari specific crashes in `Texture 3D` and `GLTF` demos.

#### Change List
- Website: demo fixes

### CODE REVIEW

File: examples/core/texture-3d/app.js [hunk 1]
```diff
@@ -158,7 +158,9 @@ export default class AppAnimationLoop extends AnimationLoop {
   }
 
   onFinalize({gl, cloud}) {
-    cloud.delete();
+    if (cloud) {
```
Comment (ibgreen) [inline examples/core/texture-3d/app.js]: Why do we need the if statement? finalize should never be called if initialize didn't complete.

File: modules/webgl/src/classes/texture.js [hunk 2]
```diff
@@ -696,7 +696,15 @@ export default class Texture extends Resource {
   }
 
   _isNPOT() {
-    return !isWebGL2(this.gl) && (!isPowerOfTwo(this.width) || !isPowerOfTwo(this.height));
+    if (isWebGL2(this.gl)) {
+      // NPOT restriction is only for WebGL1
+      return false;
+    }
+    // Width and height not available, consider it is not NPOT texture
+    if (!this.width || !this.height) {
```
Comment (tsherif) [inline modules/webgl/src/classes/texture.js]: Interesting... what was happening that required this.

File: examples/core/texture-3d/app.js (hunk 1 above)
Comment (tsherif) [reply examples/core/texture-3d/app.js]: The initialize/finalize logic needs to be rethought. I believe `onInitialize` is called once the first time `start` is called, while `onFinalize` is called everything `stop` is called.

Comment (1chandu) [reply examples/core/texture-3d/app.js]: Its tracked here: https://github.com/uber/luma.gl/issues/1068

File: modules/webgl/src/classes/texture.js (hunk 2 above)
Comment (1chandu) [reply modules/webgl/src/classes/texture.js]: texture-cube initialization is not properly setup. #1070

**Pull request merged**